src/main/java/fr/inria/coming/codefeatures/FeatureAnalyzer.java

```diff
@@ -97,14 +93,18 @@ public AnalysisResult analyze(IRevision revision, RevisionResult previousResults
 							JsonObject opjson = JSonPatternInstanceOutput.getJSONFromOperator(operation);
 							jsonFeature.add("ast_info", opjson);
 						}
-						if(p4jfeatures!=null) {
-							jsonFeature.add("P4J",p4jfeatures.getFeatures());
-						}
+						
 						changesArray.add(jsonFeature);
 					}
 				}
 				
 			}
+			
+			FeaturesResult p4jfeatures = (FeaturesResult) new P4JFeatureAnalyzer().analyze(revision, nameFile+"");		
```

Comment:
Why appends an empty string? Is it missing something (eg. file extension)?

Reply:
just convert from object type to String type, updated to nameFile.toString()